### PR TITLE
[ARCTIC-262]: Feature terminal mulit backend

### LINKED ISSUE #262: [Feature]: AMS terminal use Kyuubi as its backend SQL engine
### Description

AMS terminal use Kyuubi as its backend sql engine

### Use case/motivation

AMS terminal use a embedded spark client as its SQL engine now.
When user use termial to execute some large SQL, it may cause long executation time and other unknown errors.
Using Kyuubi as its backend SQL enging would solve this problem.

More Kyuubi information would be found here:https://kyuubi.apache.org/

### Describe the solution

- Add config to set Kyuubi as AMS termial SQL engine
- Execute user SQL with Kyuubi

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?
support mulit backend for ams terminal. Resolve #262 

## Brief change log

* add TerminalManager to manage terminal session creating and release.
* add TerminalSessionManager interface to support mulit backend
* implement TerminalSessionManager, add LocalSessionFactory as terminal backend type=local
* implement TerminalSessionManager, add KyuubiSessionFactory as terminal backend type=kyuubi

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes )
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionContext.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+          return ExecutionStatus.Canceled;
+        }
+        if (statementBuilder == null) {
+          statementBuilder = new StringBuilder();
+        }
+        line = line.trim();
+        if (line.length() < 1 || line.startsWith("--")) {
+          // ignore blank lines and comments.
+          continue;
+        } else if (line.endsWith(";")) {
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionContext.java:207]: It seems that now only one statement is allowed on a line，so maybe add some comments about that and a TODO for future improvement.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/ExecutionContext.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.server.terminal;
+
+public class ExecutionContext {
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/ExecutionContext.java:21]: It seems this class is not used?

File: dist/src/main/arctic-bin/conf/config.yaml [hunk 3]
```diff
@@ -23,7 +23,11 @@ ams:
   #arctic.ams.cluster.name: default
   #arctic.ams.zookeeper.server: 127.0.0.1:2181,127.0.0.1:2182,127.0.0.1:2183
 
-# extension properties for like system
+  # kyuubi config
```
Comment (zhoujinsong) [inline dist/src/main/arctic-bin/conf/config.yaml:26]: Should we add default local terminal configurations here?

**Review by zhoujinsong — COMMENTED** [review]
@baiyangtx Hi，I left some comments.

File: dist/src/main/arctic-bin/conf/config.yaml (hunk 3 above)
Comment (baiyangtx) [reply dist/src/main/arctic-bin/conf/config.yaml:26]: local is default in code, does not need to be setted in conf file.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+  private final AtomicLong threadPoolCount = new AtomicLong();
+  TerminalSessionFactory sessionFactory;
+  int resultLimits = 1000;
+  boolean stopOnError = false;
+
+  int sessionTimeout = 30;
+  int sessionTimeoutCheckInterval = 5;
+
+  private final Object sessionMapLock = new Object();
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java:60]: changed to hash map

**Comment by zhoujinsong** [comment]
@baiyangtx You may need add a `Resolve #262 ` content in your PR description to help github link this issue to PR and delete the `WIP` tag in your PR title when you are ready to merge this PR.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**